### ISSUE TITLE
KeyError: 'key of type tuple not found and not a MultiIndex'

### ISSUE THREAD

**Issue by alon-sht** [djwooten/synergy#40]
### 🐛 Describe the bug

df = pd.read_csv(r"sample_data_1.csv")

d1 = df['drug1.conc']
d2 = df['drug2.conc']
E = df['effect']

model = Bliss()
synergy = model.fit(d1, d2, E)

i get: KeyError: 'key of type tuple not found and not a MultiIndex'
only started in the new version of synergy...

### Versions

python 3.11.4
pandas                        1.5.3
numpy                         1.25.2
scikit-learn                  1.2.2
scipy                         1.11.2

**Comment by dipanghosh** [comment]
Yes, I am having the same issue. I will dive into the codebase and have a look. 

I just started using the library so I am not sure if this bug was introduced recently.

In addition, installing from PIP, the load_datasets_* functions are not working as the csv data files are just not present.

**Comment by dipanghosh** [comment]
Looked into it. The data needs to be loaded correctly. This is not _necessarily_ a bug, but I am guessing that the documentation needs to be updated. 

Make sure to convert your d1, d1 and E to a numpy array.

```
d1 = df['drug1.conc'].to_numpy()
d2 = df['drug2.conc'].to_numpy()
E = df['effect'].to_numpy()
```

Then I got past the initial error, but now I am getting `'numpy.float64' object has no attribute 'log'`

Turns out one the columns had the wrong datatype, so be sure to cast it to a float. The library will not do this, and the errors do not say why the error was.

```
d1 = df['drug1.conc'].to_numpy().astype(float)
d2 = df['drug2.conc'].to_numpy().astype(float)
E = df['effect'].to_numpy().astype(float)
```

Finally got me there.

**Comment by abearab** [comment]
@djwooten any updates? I also a similar issue
